### PR TITLE
feat(blade): Radio Motion - PR4

### PR DESCRIPTION
**Pull request by anuraghazra**
- Added motion for native & web
- Fixed color variant issues

Prev: #615  
Next: #631 

-----

PR Stack: 

1. #605
2. #607
3. #615
4. #629 (Current)
5. #631

### CODE REVIEW

File: packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { EasingFn } from 'react-native-reanimated';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx:4]:
```suggestion
import type { EasingFunctionFactory } from 'react-native-reanimated';
```

File: packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  return {
+    ...styles,
+  };
+});
+
+const Fade = ({ children, show, styles }: FadeProps) => {
+  const { theme } = useTheme();
+
+  const enterEasing = (theme.motion.easing.entrance.effective as unknown) as EasingFn;
+  const leaveEasing = (theme.motion.easing.exit.effective as unknown) as EasingFn;
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx:21]:
```suggestion
  const enterEasing = (theme.motion.easing.entrance.effective as unknown) as EasingFunctionFactory;
  const exitEasing = (theme.motion.easing.exit.effective as unknown) as EasingFunctionFactory;
```

File: packages/blade/src/components/Checkbox/CheckboxIcon/Fade.web.tsx [hunk 3]
```diff
@@ -42,29 +44,30 @@ const AnimatedFade = styled.div<{ animationType: FlattenSimpleInterpolation | nu
...
   const { theme } = useTheme();
 
+  const duration = theme.motion.duration.xquick;
   const enter = css`
-    animation: ${fadeIn} ${makeMotionTime(theme.motion.duration.xquick)}
+    animation: ${fadeIn} ${makeMotionTime(duration)}
       ${theme.motion.easing.entrance.effective as string};
   `;
 
   const leave = css`
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Checkbox/CheckboxIcon/Fade.web.tsx:53]:
```suggestion
  const exit = css`
```

File: packages/blade/src/components/Radio/RadioIcon/Fade.web.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    animation: ${scaleIn} ${makeMotionTime(duration)}
+      ${theme.motion.easing.entrance.effective as string};
+  `;
+
+  const leave = css`
+    animation: ${fadeOut} ${makeMotionTime(duration)}
+      ${theme.motion.easing.exit.effective as string};
+  `;
+
+  const { isMounted, isVisible } = usePresence(!!show, {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/RadioIcon/Fade.web.tsx:58]: Can you add some short comments to explain whats happening here for reference?

File: packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx:4]: Throws error actually. the Keyframes() object takes EasingFn only.

**Pull request merged**